data/topic-filter.go

```diff
@@ -0,0 +1,222 @@
+package data
```

Comment:
is this a c+p of filter.go? do we need all of it? can we re-use parts from both of these to reduce duplication? can see parts here we don't need e.g. `defaultTopicContentTypes` since those are used for the `?filter=` filter not topic filter

Reply:
So I've added comment addressing renaming things related to topics and removing unnecessary duplication. We could go a step further and most of these functions could be turned into ones that handle interfaces instead of TopicFilter and ContentTypeFilter (which is described Filter everywhere). That way there would be even less duplication and easier to extend later down the line when another filter option becomes available.

I suggest this can be looked at in a separate ticket.